### ISSUE TITLE
<model>ForQuestionAnswering loading non-deterministic weights

### ISSUE THREAD

**Issue by aychang95** [huggingface/pytorch-transformers#949]
I was comparing the weight and bias parameters of two different pre-trained-loaded BertForQuestionAnswering model, and they seem to differ.  This causes every instantiation of pre-trained models to have slightly different results.

Compared to #695 where you set the model to eval mode to deactivate dropout layers, the non-deterministic trait seems to come from loading pre-trained models with `BertForQuestionAnswering.from_pretrained("bert-base-uncased")`

To replicate what I'm talking about, you can see below.
```python
model_1 = BertForQuestionAnswering.from_pretrained("bert-base-uncased")
model_2 = BertForQuestionAnswering.from_pretrained("bert-base-uncased")

weights_1 = model_1.state_dict()['qa_outputs.weight']
weights_2 = model_2.state_dict()['qa_outputs.weight']

torch.eq(weights_1, weights_2)

```
This also occurs in XLNetForQuestionAnswering and I was curious to how/why this works that way?

**Comment by thomwolf** [comment]
These weights are not pretrained, they are added for fine-tuning the model on a downstream question answering task. You have to train the `qa_output` weights.
They are initialized randomly and so will be different at each run.
